cirq/devices/noise_model.py

```diff
@@ -83,14 +78,15 @@ def noisy_moment(self, moment: 'cirq.Moment',
         Returns:
             An OP_TREE corresponding to the noisy operations for the moment.
         """
-        if not hasattr(self.noisy_moments, '_not_overridden'):
+        if self._alternative_for('noisy_moment') == 'noisy_moments':
```

Comment:
Is it possible to resolve this at init time, instead of during every invocation?